main.c

```diff
@@ -29,57 +33,95 @@ void printString (char* str) {
 }
 
 void saveCursorPos () {
-    printf("%c7", 27);
-    fflush(stdout);
+    printSpecialChar("7");
 }
 
 void restoreCursorPos () {
-    printf("%c8", 27);
-    fflush(stdout);
+    printSpecialChar("8");
+}
+
+void redrawPrevLines (char matrix[1000][1000], int currentLine)
+{
+  int i, j;
```

Comment:
i and j don't have to be local variables, please declare them as part of the `for` cycle, just like you did in other places.

Reply:
Done